ckanext/xloader/utils.py

```diff
@@ -85,18 +85,32 @@ def resource_data(id, resource_id):
 
 
 def get_xloader_user_apitoken():
+    # type: () -> str
     """ Returns the API Token for authentication.
 
+    xloader downloads files and the API key is needed for private resources.
+    """
+    user = p.toolkit.config.get('ckanext.xloader.user', None)
+    if user and user.get('apikey'):
+        return user.get('apikey')
+
+    user = p.toolkit.get_action('get_site_user')({'ignore_auth': True}, {})
+    return user["apikey"]
+
+
+def get_xloader_user_context():
+    # type: () -> Context|dict
+    """ Returns the Xloader user.
+
     xloader actions require an authenticated user to perform the actions. This
-    method returns the api_token set in the config file and defaults to the
-    site_user.
+    method returns the context for actions.
     """
-    api_token = p.toolkit.config.get('ckanext.xloader.api_token', None)
-    if api_token:
-        return api_token
+    user = p.toolkit.config.get('ckanext.xloader.user', None)
+    if user:
+        return {"user": user['name']}
```

Comment:
`ckanext.xloader.user` config is a dict?

Reply:
Also oops, yeah I meant this to just be a username but was copying sysadmin code. This is fixed now.